### ISSUE TITLE
Treeview is not accessible using NVDA

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#6667]
The is an issue collecting known problems using the NVDA screen reader on the experimental Treeview component.

## Environment

NVDA on Firefox latest in Windows 10

## Issues:
- [ ] Selected listitems are not described as 'selected' when they receive focus
  _Selected list items should read as 'selected' when they receive focus_

- [ ] Hitting enter or space bar selects the focused list item and does not toggle open/close when nested lists are present
  _The expected on click event for a tree view node with a sub-node is to toggle open/close state on `enter` or `space`_

- [ ] Disabled lists are not accessible
  _If the disabled state of a list item is visible to sighted users that state should be described to screen reader users as well_
